### PR TITLE
Module updates, fix Security.getSchemaPermissions

### PR DESCRIPTION
**Pull request by labkey-nicka**
#### Related Pull Requests
* https://github.com/LabKey/platform/pull/1078 (version bump)

#### Changes
* Module updates for `Exp`, `Message`, `Pipeline`, and `Report`.
* Fix for `Security.getSchemaPermissions()` to defensively copy configuration for `getSecurableResources` call.

### CODE REVIEW

File: src/labkey/Pipeline.ts [hunk 1]
```diff
@@ -48,13 +49,13 @@ export interface IGetFileStatusOptions {
      * - name: name of the file, a String.
      * - status: status of the file, a String
      */
-    success?: () => any
+    success?: RequestSuccess
```
Comment (cnathe) [inline src/labkey/Pipeline.ts:52]: I assume there is a reason that this success can't just be removed and come from RequestCallbackOptions.

File: src/labkey/Pipeline.ts [hunk 2]
```diff
@@ -96,25 +88,23 @@ export interface IGetPipelineContainerOptions {
      * been defined in this container hierarchy, the value of this property will be null.
      * - webDavURL: the WebDavURL for the pipeline root.
      */
-    success?: Function
+    success?: RequestSuccess
```
Comment (cnathe) [inline src/labkey/Pipeline.ts:91]: Same comment as above, any reason this success can't be removed and come from RequestCallbackOptions?

File: src/labkey/Pipeline.ts [hunk 3]
```diff
@@ -143,13 +125,13 @@ export interface IGetProtocolsOptions {
      * - jsonParameters: JSON representation of the parameters defined by this protocol.
      * - containerPath: The container path where this protocol was saved
      */
-    success?: Function
+    success?: RequestSuccess
```
Comment (cnathe) [inline src/labkey/Pipeline.ts:128]: same comment

File: src/labkey/Pipeline.ts (hunk 1 above)
Comment (labkey-nicka) [reply src/labkey/Pipeline.ts]: Thanks for having me take a second look at this. Turns out not even `RequestSuccess` is correct in a couple of these cases. A couple of the success callbacks have "custom" function signatures which would not be properly annotated by `RequestSuccess`. Due to this I had to not use `RequestCallbackOptions` at all in these scenarios.

See https://github.com/LabKey/labkey-api-js/pull/53/commits/b7ad4d954b92df165aed88005367860acfdaed78.

File: src/labkey/Pipeline.ts (hunk 2 above)
Comment (labkey-nicka) [reply src/labkey/Pipeline.ts]: See comment above.

File: src/labkey/Pipeline.ts (hunk 3 above)
Comment (labkey-nicka) [reply src/labkey/Pipeline.ts]: See comment above.

**Pull request merged**